pilot/pkg/networking/util/util_test.go

```diff
@@ -532,3 +532,62 @@ func TestIsHTTPFilterChain(t *testing.T) {
 		t.Errorf("tcp filter chain detected as http filter chain")
 	}
 }
+
+var (
+	listener80 = &v2.Listener{Address: BuildAddress("0.0.0.0", 80)}
+	listener81 = &v2.Listener{Address: BuildAddress("0.0.0.0", 81)}
+	listenerip = &v2.Listener{Address: BuildAddress("1.1.1.1", 80)}
+)
+
+func BenchmarkGetByAddress(b *testing.B) {
+	for n := 0; n < b.N; n++ {
+		GetByAddress([]*v2.Listener{
+			listener81,
+			listenerip,
+			listener80,
+		}, listener81.Address)
```

Comment:
for the benchmark, should we pass in the last address for a worst case scenario?